Automattic/woocommerce-payments
# Add progressive onboarding mode choice

Fixes #5480 

#### Changes proposed in this Pull Request
- Add required icon as JSX components to allow color adjustment through CSS.
- Implement reusable component `TipBox` with `purple` and `blue` colors.
  - Not needed anymore, but I'll leave it there as it might be required soon.
- Implement reusable component `RadioCard`.
- Add missing color variables to `_variables.scss`.
- Add `ModeChoice` step matching design requirements.
- Update `WC_Payments_Onboarding_Service` to hook into `wcpay_dev_mode` and enable it based on the new option `wcpay_onboarding_test_mode`.
- Enable onboarding test mode in `init_stripe_onboarding` based on `$_GET['test_mode']`.

To let merchants onboard and use a test account we need to enable `dev_mode`. Right now we offer them the option to do it using `WCPAY_DEV_MODE` constant. To allow it to be enabled while onboarding I've chosen to add a new option, and filter `dev_mode` based on it.

#### Testing instructions
- Ensure you're not connected to an account (for example, you can mark it as deleted in the DB)
- Ensure `Dev mode enabled` is unchecked and `Enable Progressive Onboarding` is checked in `WCPay Dev` plugin.
- Go to **Payments** and click on **Get started**
- The live option should be selected and clicking on **Continue** should move to the next step (personal details form).
- Refresh, choose the test option and click on **Continue**.
- You should be redirected to Stripe full KYC in test mode, complete it.
- After being redirected back, the site should have a fully functional test account onboarded.

#### Images
| Live | Test|
|-|-|
|<img width="418" alt="Screenshot 2023-03-31 at 11 53 11" src="https://user-images.githubusercontent.com/7670276/229088356-fbe9c512-37bd-4e51-a785-212675931a87.png">|<img width="420" alt="Screenshot 2023-03-31 at 11 53 25" src="https://user-images.githubusercontent.com/7670276/229088345-0a02cf73-b9d5-4466-adfa-3cd8751071b9.png">|

-------------------


**Post merge**
